### DISCUSSION TITLE
Remove deprecated methods in data providers and models

### DISCUSSION THREAD

**Discussion by hmlnarik** [keycloak/keycloak#13565] [Ideas]
## Description

Since 2020, multiple methods became deprecated, usually replaced by a similar method with a consistent parameter positions and / or stream variants. Examples of these transitions are below.

I propose **removing all of these deprecated methods in Keycloak 20**. Maintaining the deprecated methods brings no value, and the deprecation period has been IMO sufficient.

Please state your agreement or concerns with this intention by Sep 15 here.

Cc: @keycloak/kc-maintainers @keycloak/kc-external-maintainers 

### Scope

All `*Model` classes and all data-oriented `*Provider` classes like `UserProvider`, `GroupProvider` etc.

### Examples of these methods

Remove `getRequiredCredentials()` here:
https://github.com/keycloak/keycloak/blob/4323d796b392f9fc7a39702cc41ac98af7295b98/server-spi/src/main/java/org/keycloak/models/RealmModel.java#L270-L282

Remove `searchForGroupByName()` here:
https://github.com/keycloak/keycloak/blob/4323d796b392f9fc7a39702cc41ac98af7295b98/server-spi/src/main/java/org/keycloak/storage/group/GroupLookupProvider.java#L37-L65

**Comment by mduchrow** [comment]
As long as there is an adequate replacement method (and I think it is) the removal should be no problem.
Perhaps a hint the release-notes of KC 20 that the replacements for deprecated methods are documented in the KC 19 code could help with the migration of plugins.

**Comment by abstractj** [comment]
@hmlnarik what you suggest make sense. I'd suggest a blog post with all the details about which APIs will become EOL by Sept 15, so it's transparent to our community. Here's an example from Node.js https://nodejs.org/api/deprecations.html

**Reply by hmlnarik** [reply]
+1 to the blog post, this must be needs to be well advertised.
Re timing, by Sep 15, I would like to have go / no-go decision. The blog would be published shortly afterwards.

**Reply by abstractj** [reply]
+1 from me

**Comment by mposolda** [comment]
+1 for remove these deprecated methods. The 2 years period should be enough and there was already multiple Keycloak releases with the deprecated methods.
